src/data/finset/pointwise.lean

```diff
@@ -393,6 +393,18 @@ begin
     exact ih.trans (subset_mul_right _ hs) }
 end
 
+@[simp, norm_cast, to_additive]
+lemma coe_list_prod (s : list (finset α)) : (↑s.prod : set α) = (s.map coe).prod :=
+map_list_prod (coe_monoid_hom : finset α →* set α) _
+
+@[to_additive] lemma mem_prod_list_of_fn {a : α} {s : fin n → finset α} :
+  a ∈ (list.of_fn s).prod ↔ ∃ f : (Π i : fin n, s i), (list.of_fn (λ i, (f i : α))).prod = a :=
+by { rw [←mem_coe, coe_list_prod, list.map_of_fn, set.mem_prod_list_of_fn], refl }
+
+@[to_additive] lemma mem_pow {a : α} {n : ℕ} :
+  a ∈ s ^ n ↔ ∃ f : fin n → s, (list.of_fn (λ i, ↑(f i))).prod = a :=
```

Comment:
Wait, isn't it true that `(list.of_fn f).prod = ∏ i, f i`? If so, then
```suggestion
  a ∈ s ^ n ↔ ∃ f : fin n → s, ∏ i, ↑(f i) = a :=
```
and please write the simp lemma `prod_list_of_fn`.

Reply:
Not when the monoid is non-commutative, which is the whole reason I need this

Reply:
Ah yeah sorry, of course.

Reply:
Ah yeah sorry, of course.